lib/chef-dk/policyfile/undo_stack.rb

```diff
@@ -0,0 +1,98 @@
+#
+# Copyright:: Copyright (c) 2015 Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'fileutils'
+
+require 'ffi_yajl'
+
+require 'chef-dk/helpers'
+require 'chef-dk/policyfile/undo_record'
+
+module ChefDK
+  module Policyfile
+
+    class UndoStack
+
+      MAX_SIZE = 10
```

Comment:
The use case I'm trying to address is the "oh crap I didn't mean to delete *that* thing" or "oops I thought no one was using that" kind of thing. Think of this as an alternative to having a confirmation dialog (e.g., "are you sure you want to delete X?"). Since it's just stored locally on your laptop, it won't be useful to other folks on your team, so I don't think it can serve a generalized backup function (need to use DB snapshots or knife EC backup instead).

Make sense?